docs/guides/file-system-based-automated-bundle-generation.md

```diff
@@ -0,0 +1,116 @@
+
+```gitignore
+### Generated React on Rails packs
+packs/generated
+```
+
+*Note, the directory might be different depending on the `source_entry_path` in `config/webpacker.yml`.*
+
+## Usage
+
+### Basic usage
+
+Considering the `webpacker.yml` file is configured as instructed [here](https://github.com/shakacode/shakapacker#configuration-and-code), With following configurations
+
+```yml
+default: &default
+source_path: app/javascript
+source_entry_path: packs 
+public_root_path: public
+public_output_path: packs
+nested_entries: true
+# And more
+```
+
+the directory structure will look like this
+```
+app/javascript:
+  └── packs:               # sets up webpack entries
+  │   └── application.js   # references ../src/my_component.js
+  │   └── application.css
+  └── src:                 # any directory name is fine. Referenced files need to be under source_path
+  │   └── component.js
+  └── stylesheets:
+  │   └── my_styles.css
+  └── images:
+      └── logo.svg
+public/packs                # webpack output
+```
+
+Now, To automatically register, `A_ComponentOne`, `B_ComponentOne`, `B_ComponentTwo` for the usage with `react_component` & `react_component_hash` helpers, Create directory structure as mentioned below:
+
+```
+app/javascript:
+  └── packs:                   
+  │   └── application.js     
+  │   └── application.css
+  └── src:                   
+  │   └── A
+  │   │ └── ...
+  │   │ └── ror_components          # configured as `components_directory`
+  │   │   └── A_ComponentOne.jsx
+  │   └── B
+  │   │ └── ror_components          # configured as `components_directory`
+  │   │   │ └── B_ComponentOne.jsx
+  │   │   │ └── B_ComponentTwo.jsx
+public/packs                
+```
+
+To register react component, It is no longer needed to create a pack entry and manually register it by calling `ReactOnRails.register`. With automatically generated packs, you can directly use `A_ComponentOne`, `B_ComponentOne` and `B_ComponentTwo` in rails view using:
+
+```erb
+    <%= react_component("A_ComponentOne", {}, auto_load_bundle: true) %>    
+    <%= react_component("B_ComponentOne", {}, auto_load_bundle: true) %>    
+    <%= react_component("B_ComponentTwo", {}, auto_load_bundle: true) %>    
+```
+
+`auto_load_bundle` option can be omitted when, if `config/initializers/react_on_rails` is configured with `config.auto_load_bundle = true`.
+
+### Server Rendering and Client Rendering Components
+
+If server rendering is enabled, the component will be registered for the usage both on server and client rendering. In order to have separate definitions for client and server rendering, Name the components files as `Component_Name.server.jsx` and `Component_Name.client.jsx`. During the server rendering, the `Component_Name.server.jsx` definition will be used, and for client rendering, `Component_Name.client.jsx` file will be used.
```

Comment:
When is the simple `Component_Name.jsx` supposed to be used? From this description it looks like you have two options: 
1. If you want rendering to be the same, use just `Component_Name.jsx`, no `Component_Name.server.jsx` and no `Component_Name.client.jsx`.
2. If you want rendering to be different, use `Component_Name.server.jsx` and `Component_Name.client.jsx`, no `Component_Name.jsx`. 

But then why do we have tests for mixing them? Should such mixing raise an error instead?

Reply:
Again, not resolved, and I think this is important.

Reply:
"Name the components files" shouldn't be capitalized, and "component files".